services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -0,0 +1,955 @@
+            If we failed to get the external files sizes to make sure we can convert the dataset to parquet
+        - [`~job_runners.config.parquet_and_info.ExternalFilesSizeRequestTimeoutError`]
+            If we failed to get the external files sizes to make sure we can convert the dataset to parquet
+        - [`~job_runners.config.parquet_and_info.ExternalFilesSizeRequestError`]
+            If we failed to get the external files sizes to make sure we can convert the dataset to parquet
+    </Tip>
+    """
+    logging.info(f"get parquet files and dataset info for {dataset=} {config=}")
+
+    raise_if_not_supported(
+        dataset=dataset,
+        config=config,
+        hf_endpoint=hf_endpoint,
+        hf_token=hf_token,
+        committer_hf_token=committer_hf_token,
+        revision=source_revision,
+        supported_datasets=supported_datasets,
+        blocked_datasets=blocked_datasets,
+        max_dataset_size=max_dataset_size,
+    )
+
+    hf_api = HfApi(endpoint=hf_endpoint, token=hf_token)
+    committer_hf_api = HfApi(endpoint=hf_endpoint, token=committer_hf_token)
+
+    # prepare the parquet files locally
+    local_parquet_files: List[ParquetFile] = []
+    download_config = DownloadConfig(delete_extracted=True)
+    try:
+        builder = load_dataset_builder(
+            path=dataset,
+            name=config,
+            revision=source_revision,
+            use_auth_token=hf_token,
+            download_config=download_config,
+        )
+    except _EmptyDatasetError as err:
+        raise EmptyDatasetError(f"{dataset=} is empty.", cause=err) from err
+    raise_if_too_big_from_external_data_files(
+        builder=builder,
+        max_dataset_size=max_dataset_size,
+        max_external_data_files=max_external_data_files,
+        hf_token=hf_token,
+    )
+    builder.download_and_prepare(file_format="parquet")  # the parquet files are stored in the cache dir
+    dataset_info = asdict(builder.info)
+    local_parquet_files.extend(
+        ParquetFile(local_file=local_file, local_dir=builder.cache_dir, config=config)
+        for local_file in glob.glob(f"{builder.cache_dir}**/*.parquet")
+    )
+
+    # create the target revision if it does not exist yet
+    try:
+        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
+        if all(ref.ref != target_revision for ref in refs.converts):
+            committer_hf_api.create_branch(
+                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=source_revision
+            )
+    except RepositoryNotFoundError as err:
+        raise DatasetNotFoundError("The dataset does not exist on the Hub.") from err
+
+    target_dataset_info = hf_api.dataset_info(repo_id=dataset, revision=target_revision, files_metadata=False)
+    # - check if there are files for configs that do not exist anymore and delete them
+    # - get dataset's config names with `datasets` lib directly
```

Comment:
I first did [like this](https://github.com/huggingface/datasets-server/pull/985/commits/e8adfff58d5a1944fc3bd95edd532162f2490d1d) - try from cache first, then with the `datasets` lib if cache response is not successful. Then I thought that the cache might be outdated(???)... But it's actually not possible because we run `config-parquet-and-info` only after `/config-names`, right? So it should be taken only from cache?

Reply:
[af9df65](https://github.com/huggingface/datasets-server/pull/985/commits/af9df651fe4a2335c7afe54d613afefa479c48a9)